### ISSUE TITLE
ToD DC Correction and a couple additions.

### ISSUE THREAD

**Issue by CeruleanSky** [Sora3100/Tales_of_Cheat_Tables#80]
I am working my way through the game and thought I could contribute some corrections and additions I make as I do so.  Thank you for providing this cheat table.   

Pull requests are hard to parse and messy on cheat tables sometimes because of how cheat table xml is generated so I am attaching the file and inlining the changes to this issue instead.  Should just be able to copy the code and past it into the cheat engine interface.

Just in case I wonder off I'll leave what I have so far, if I end up with more I'll either edit this comment or add a new comment.

Enemy Groups needed to be 2 bytes long.   It was really confusing seeing late at night battle group `262` as `6` lol.
<details>
  <summary>Enemy Group Mod</summary>  

```<?xml version="1.0" encoding="utf-8"?>
<CheatTable>
  <CheatEntries>
    <CheatEntry>
      <ID>17325</ID>
      <Description>"Enemy Group Mod"</Description>
      <DropDownList DisplayValueAsItem="1">1:Stahn
4:Seinegald Soldiers
5:Seinegald Soldiers (2nd part)
6:Leon (1st encounter)
8:Basilisk x 5
9:Kongman (1st Colosseum fight)
11:Batista
12:Cave Queen
13:Batista Diego
14:Kraken
15:Tiberius
17:Dalis
18:Lydon
19:Leon (Cave Fight)
20:Force Armor (Stahn Side)
21:Baruk
22:Irene
23:Rembrandt
24:Hugo
25:Miktran (1st encounter)
26:Buer
29:Dymlos
31:Shadow of Stahn
32:Shadow of Woodrow
33:Shadow of Stahn &amp; Leon
34:Kronos
35:Kronos Excessive
36:Kraken
43:Kongman (2nd Colosseum fight)
44:Kongman (Female)
45:Lilith (Colosseum)
46:Lilith (Retry version?)
48:Floor 1 (EX Dungeon)
49:Cupid (EX Dungeon)
50:Rune Hearts (EX Dungeon)
51:Molflina Bunny (EX Dungeon)
52:Finley (EX Dungeon)
53:Super Star (EX Dungeon)
54:Empire Beetle (EX Dungeon)
55:KOZ-MOZ Gentleman (EX Dungeon)
56:Masalina Bunny (EX Dungeon)
57:Barbatos (EX Dungeon)
58:3 Thieves
60:Sidequest boss?
61:Barbatos (anti grinding)
62:Ras Gentleman
63:Klarth Gentleman
64:Hilda Gentleman
65:Chloe Gentleman
66:Ras &amp; Klarth &amp; Hilda &amp; Chloe Gentleman
67:Lens Hunters (Stahn &amp; Rutee &amp; Mary)
68:Swordian Masters (Stahn &amp; Rutee &amp; Woodrow &amp; Philia)
71:2 Star Fish
75:Force Armor (Leon Side)
80:Dycroft Enemies
</DropDownList>
      <LastState Value="0" RealAddress="7FF6402C4A44"/>
      <ShowAsSigned>0</ShowAsSigned>
      <VariableType>2 Bytes</VariableType>
      <Address>baseElfAddr+1C49C8</Address>
    </CheatEntry>
  </CheatEntries>
</CheatTable>

```
  
</details>

In `Other` added `World Coordinates`  I suggest renaming the existing `Coordinates` code to `Map coordinates`. This allowed me to travel back to Seingard Castle early in the game even after I left the continent to do finish an event I missed.

<details>
  <summary>World Coordinates</summary>  

  ```<?xml version="1.0" encoding="utf-8"?>
<CheatTable>
  <CheatEntries>
    <CheatEntry>
      <ID>20931</ID>
      <Description>"World Coordinates"</Description>
      <Options moHideChildren="1"/>
      <LastState Value="" RealAddress="7FF64043F080"/>
      <ShowAsSigned>0</ShowAsSigned>
      <GroupHeader>1</GroupHeader>
      <Address>baseElfAddr+33F004</Address>
      <CheatEntries>
        <CheatEntry>
          <ID>20932</ID>
          <Description>"X"</Description>
          <LastState Value="0" RealAddress="7FF64043F080"/>
          <ShowAsSigned>0</ShowAsSigned>
          <VariableType>Float</VariableType>
          <Address>+0</Address>
        </CheatEntry>
        <CheatEntry>
          <ID>20933</ID>
          <Description>"Y"</Description>
          <LastState Value="0" RealAddress="7FF64043F084"/>
          <ShowAsSigned>0</ShowAsSigned>
          <VariableType>Float</VariableType>
          <Address>+4</Address>
        </CheatEntry>
        <CheatEntry>
          <ID>20934</ID>
          <Description>"Z"</Description>
          <LastState Value="0" RealAddress="7FF64043F088"/>
          <ShowAsSigned>0</ShowAsSigned>
          <VariableType>Float</VariableType>
          <Address>+8</Address>
        </CheatEntry>
      </CheatEntries>
    </CheatEntry>
  </CheatEntries>
</CheatTable>

  ```
  
</details>

The number of hits in battle and the max hits achieved in the battle.

<details>
  <summary>In Battle Combo Hits</summary>  

```<?xml version="1.0" encoding="utf-8"?>
<CheatTable>
  <CheatEntries>
    <CheatEntry>
      <ID>20915</ID>
      <Description>"Current Combo Hits"</Description>
      <LastState Value="0" RealAddress="7FF640422154"/>
      <ShowAsSigned>0</ShowAsSigned>
      <VariableType>Byte</VariableType>
      <Address>baseElfAddr+3220D8</Address>
    </CheatEntry>
    <CheatEntry>
      <ID>20916</ID>
      <Description>"Max Combo Hits"</Description>
      <LastState Value="0" RealAddress="7FF640422160"/>
      <ShowAsSigned>0</ShowAsSigned>
      <VariableType>Byte</VariableType>
      <Address>baseElfAddr+3220E4</Address>
    </CheatEntry>
  </CheatEntries>
</CheatTable>

```
  
</details>

[Tales of Destiny_DC.zip](https://github.com/Sora3100/Tales_of_Cheat_Tables/files/15241051/Tales.of.Destiny_DC.zip)

Bonus:
Pnach script for infinite item usage, I didn't get this working in cheat engine, maybe because of how the recompiler works or I have a setting wrong.

```
[InfiniteItems]
author=anonymous
description=items do not decrease
patch=1,EE,20172790,extended,DC120000
```

<details>
<summary>How it was done for those curious</summary>
I used the pcsx2 debuger, found memory address that decreased, breakpointed on writes to that items count, then ghidraed + [ghidra-emotionengine-reloaded](https://github.com/chaoticgd/ghidra-emotionengine-reloaded) extension the routine at that memory address to study the code and altered it so that it adds zero to the number of items when decrementing. The "EE_Core_Instruction_Set_Manual" was helpful for figuring that out.
</details>

**Comment by Sora3100** [comment]
Thank you, these will get added the next time I update the tables.

**Comment by CeruleanSky** [comment]
As I am in the last quarter in the game, I'll probably at most comment once more if I have anything else, but the table for the most part at least for me has had everything I've needed.

Stats->Character-> Paste it into each character

<details><summary>Support Skill Countdown</summary>

```
<?xml version="1.0" encoding="utf-8"?>
<CheatTable>
  <CheatEntries>
    <CheatEntry>
      <ID>20983</ID>
      <Description>"Support Skill Countdown"</Description>
      <LastState Value="1" RealAddress="7FF8002C2CB8"/>
      <ShowAsSigned>0</ShowAsSigned>
      <VariableType>4 Bytes</VariableType>
      <Address>+14C</Address>
    </CheatEntry>
  </CheatEntries>
</CheatTable>
```

<details>

**Comment by CeruleanSky** [comment]
Just wanted to let you know I finished the game and the previous comment has been edited to include all the codes I found useful. Feel free to add them on your next update any that you find helpful.   Thanks again for the table.  Now looking forward to Tales of Destiny 2 whenever Lumina Tales releases it.

**Comment by Sora3100** [comment]
Added everything from here with commit #84
Thanks!
